.github/workflows/tests.yml

```diff
@@ -494,6 +494,240 @@ jobs:
           name: lxd-clients-${{ runner.os }}
           path: bin/
 
+  ui:
```

Comment:
please can you move this as part of the `system-tests:` workflow and make it a step after `name: "Run system tests (${{ matrix.suite }}, ${{ matrix.backend }})"` and before `name: Upload coverage data`

Then we can re-use the dependency install deps, and by ensuring LXD is run with the code coverage env var, we can collect coverage data from the UI tests too.

Reply:
This will slow down LXD test runtime, currently it runs in parallel to the system tests, which will not affect your total runtime. Because system tests take longer than the UI tests. I think a better option is to also record coverage in the UI step and download both coverage reports in the TICS step.

Reply:
You can still run the test itself concurrently, as long as it happens after the LXD binary is built with code coverage, and before the code coverage collection. This is what we do with the system tests.

Reply:
Adjusted the workflow, so the UI tests reuse the binaries and also upload coverage with a matching prefix. They now run in parallel to the system tests, and block the TICS run, which consumes the coverage from both.

![image](https://github.com/user-attachments/assets/30697025-07ad-4d82-8439-2f7cfaae1f3a)

Reply:
Excellent!